tests/kernel/queue/src/test_queue_contexts.c

```diff
@@ -28,13 +28,19 @@ static struct k_thread tdata;
 static K_THREAD_STACK_DEFINE(tstack1, STACK_SIZE);
 static struct k_thread tdata1;
 static struct k_sem end_sema;
+extern ZTEST_BMEM volatile bool valid_fault;
 
 void k_sys_fatal_error_handler(unsigned int reason, const z_arch_esf_t *pEsf)
 {
 	if (reason != K_ERR_KERNEL_OOPS) {
 		printk("Wrong error type\n");
 		k_fatal_halt(reason);
 	}
+	printk("Caught system error -- reason %d %d\n", reason, valid_fault);
+	if (valid_fault) {
+		printk("Fatal error expected as part of test case.\n");
+		valid_fault = false; /* reset back to normal */
+	}
```

Comment:
What will happen if the "reason==K_ERR_KERNEL_OOPS" and "valid_fault==false"?

Reply:
It seems that it will do nothing in this function.